symengine/lib/symengine_wrapper.pyx

```diff
@@ -4316,99 +4330,29 @@ def has_symbol(obj, symbol=None):
                 deref(symengine.rcp_static_cast_Symbol(s.thisptr)))
 
 
-cdef size_t _size(n):
-    try:
-        return n.size
-    except AttributeError:
-        return len(n)  # e.g. array.array
-
-
-def _get_shape_nested(ndarr):
-    # no checking of shape consistency is done
-    if isinstance(ndarr, (list, tuple)):
-        return (len(ndarr),) + _get_shape_nested(ndarr[0])
-    else:
-        return ()
-
-
-def get_shape(ndarr):
-    try:
-        return ndarr.shape
-    except AttributeError:
-        return _get_shape_nested(ndarr)
-
-
-def _nested_getitem(ndarr, indices):
-    if len(indices) == 0:
-        return ndarr
-    else:
-        return _nested_getitem(ndarr[indices[0]], indices[1:])
-
-
-def all_indices_from_shape(shape):
-    return itertools.product(*(range(dim) for dim in shape))
-
-
-def _ravel_nested(ndarr):
-    return [_nested_getitem(ndarr, indices) for indices in
-            all_indices_from_shape(get_shape(ndarr))]
-
-
-def ravel(ndarr):
-    try:
-        return ndarr.ravel()
-    except AttributeError:
-        try:
-            return _ravel_nested(ndarr.tolist())
-        except AttributeError:
-            return _ravel_nested(ndarr)
-
-
 cdef class _Lambdify(object):
-    """
-    Lambdify instances are callbacks that numerically evaluate their symbolic
-    expressions from user provided input (real or complex) into (possibly user
-    provided) output buffers (real or complex). Multidimensional data are
-    processed in their most cache-friendly way (i.e. "ravelled").
-
-    Parameters
-    ----------
-    args: iterable of Symbols
-    \*exprs: array_like of expressions
-        the shape of exprs is preserved
-    real : bool
-        Whether datatype is ``double`` (``double complex`` otherwise).
-
-    Returns
-    -------
-    callback instance with signature f(inp, out=None)
-
-    Examples
-    --------
-    >>> from symengine import var, Lambdify
-    >>> var('x y z')
-    >>> f = Lambdify([x, y, z], [x+y+z, x*y*z])
-    >>> f([2, 3, 4])
-    [ 9., 24.]
-    >>> out = np.array(2)
-    >>> f(x, out); out
-    [ 9., 24.]
-
-    """
     cdef size_t args_size, tot_out_size
     cdef list out_shapes
     cdef readonly bint real
     cdef readonly int n_exprs
+    cdef public str order
     cdef vector[int] accum_out_sizes
     cdef object numpy_dtype
+    cdef args
+    cdef tuple exprs
 
-    def __cinit__(self, args, *exprs, cppbool real=True):
+    def __cinit__(self, args, *exprs, cppbool real=True, order='C'):
```

Comment:
How about moving all the stuff from here to `__init__` and remove storing `args` and `exprs`?

Reply:
Nice catch. Works locally, pushed now.